nixos/modules/system/boot/networkd.nix

```
@@ -513,7 +513,7 @@ let
         (assertValueOneOf "EmitLLDP" (boolValues ++ ["nearest-bridge" "non-tpmr-bridge" "customer-bridge"]))
         (assertValueOneOf "DNSDefaultRoute" boolValues)
         (assertValueOneOf "IPForward" (boolValues ++ ["ipv4" "ipv6"]))
-        (assertValueOneOf "IPMasquerade" boolValues)
+        (assertValueOneOf "IPMasquerade" ["ipv4" "ipv6" "both" "no"])
```

taken from the `systemd.network` manpage:

> Note. Any positive boolean values such as "yes" or "true" are now deprecated. Please use one of the values in the above.

that reads like the values are still allowed. we probably shouldn't be more restrictive than systemd here until systemd fully removes the deprecated values.
maybe?
```suggestion
        (assertValueOneOf "IPMasquerade" (boolValues ++ ["ipv4" "ipv6" "both"]))
```

---

Yes, it makes sense.